### ISSUE TITLE
Pagefind not seeing data-pagefind-body attribute in Astro build published on Netlify

### ISSUE THREAD

**Issue by justsilencia** [CloudCannon/pagefind#163]
I have a site built with Astro, deployed with Netlify, on which I'm utilizing the pagefind library.   I have set the `data-pagefind-body` attribute on the  article element of my blog, like so:

```
<article data-pagefind-body>
       ...stuff
</article>
```

However, when the pagefind binary runs on Netlify, I get the following output:

```
[Parsing files]
**Did not find a data-pagefind-body element on the site.**
9:53:25 AM: ↳ Indexing all <body> elements on the site.
9:53:25 AM: 3 pages found without an <html> element.
9:53:25 AM: Pages without an outer <html> element will not be processed by default.
9:53:25 AM: If adding this element is not possible, use the root selector config to target a different root element.
9:53:25 AM: [Reading languages]
9:53:25 AM: Discovered 1 language: en
9:53:25 AM: [Building search indexes]
9:53:25 AM: Total:
9:53:25 AM:   Indexed 1 language
9:53:25 AM:   Indexed 4 pages
9:53:25 AM:   Indexed 33 words
9:53:25 AM: Starting post processing
9:53:25 AM:   Indexed 0 filters
9:53:25 AM:   Indexed 0 sorts
9:53:25 AM: Finished in 0.7 seconds
9:53:26 AM: Post processing - HTML
9:53:25 AM: ​
9:53:25 AM: (build.command completed in 6.2s)
```

I have reproduced this issue in a test repository here: 

[Pagefind Troubleshooting Repository](https://github.com/justsilencia/pagefind-troubleshoot)

Also, you can check out the site deployed on Netlify here and verify that the pagefind attribute is indeed in the article tag. Just click the search button in the navbar, test it out, and you can see that pagefind is indexing the entire site.

https://pagefind-troubleshoot.netlify.app/

Perhaps I'm missing something simple here. Any help would be greatly appreciated.

**Comment by bglw** [comment]
👋 Thanks a tonne for reproducing this @justsilencia ❤️

I've tracked it down to a quirk of this site — if you [view the source on one of the blog posts](view-source:https://pagefind-troubleshoot.netlify.app/blog/posts/another-way/) and scroll to the bottom you'll see a `<script client:load>` element sitting all the way outside the `<html></html>` element. A classic case of browsers being very permissive and allowing this, while Pagefind hit a weird corner case when it hit that element.

I'm not sure if that script living there is an Astro thing, or something specific to this site. In any case, I've released [Pagefind v0.10.4](https://github.com/CloudCannon/pagefind/releases/tag/v0.10.4) with a fix, these elements floating outside the `html` will no longer affect the indexing 🙂 Let me know if that resolves it!

**Comment by justsilencia** [comment]
@bglw Well, when I inspect one of the articles in devtools I'm not seeing the `<script>` tag outside of the `<html>` tag. Rather, it's at the bottom of the `<body>` as expected. This could be a false read though, because I consolidated the scripts into one `.astro` search component. So, I kinda muddied the waters there.

My guess is I did something funky with the script placement without realizing it. Nonetheless, it's working like a charm! Thanks for the quick fix, this library is super sweet for SSG blogs.
